CONTRIBUTING.md

```diff
@@ -0,0 +1,237 @@
+# Contributing to pg_vectorize
+
+## Prerequisites
+
+- [Rust](https://www.rust-lang.org/learn/get-started) - Toolchain including `rustc`, `cargo`, and `rustfmt`
+- [PGRX](https://github.com/pgcentralfoundation/pgrx) - Rust-based PostgreSQL extension development framework
+- [Docker Engine](https://docs.docker.com/engine/install/) - For running local containers
+- [psql](https://www.postgresql.org/docs/current/app-psql.html) - Terminal-based front-end to PostgreSQL
+- [pgmq](https://github.com/tembo-io/pgmq) - PostgreSQL extension for message queues
+- [pg_cron](https://github.com/citusdata/pg_cron) - PostgreSQL extension for cron-based job scheduling
+
+## Building from source
+
+This process is more involved, but can easily be distilled down into a handful of steps.
+
+### 1. Set up pgrx
+
+```bash
+cargo pgrx init
+```
+
+### 2. Set up Docker container
+
+```bash
+docker run -d -p 3000:3000 quay.io/tembo/vector-serve:latest
+```
+
+Confirm a successful set up by running the following:
+
+```bash
+docker ps
+```
+
+:wrench: Note: Consider running the following to see the container logs real time:
+
+```bash
+docker logs <your-container-id> -f
+```
+
+### 3. Clone and compile `pg_vectorize` and extension dependencies
+
+:wrench: When progressing through these steps, refer to the following for troubleshooting:
+
+```bash
+cat ~/.pgrx/15.log
+```
+
+#### 3.1. Apply configurations
+
+Prior to compiling and running `pg_vector`, it's essential to update the `postgresql.conf` file.
+`pgrx` uses a Postgres version-specific data directory, each containing its own `postgresql.conf` file.
+The following example, utilizes Postgres version 15.
+If you're using a different version, please alter the file path value `data-<postgres-version>` and run the following:
+
+```bash
+<your-editor> ~/.pgrx/data-15/postgresql.conf
+```
+
+Within this document, add the following:
+
+```text
+shared_preload_libraries = 'pg_cron, vectorize'
+cron.database_name = 'postgres'
+vectorize.embedding_service_url = 'http://vector-serve:3000/v1/embeddings'
```

Comment:
`vector-serve` will only work as the host if there's DNS set up for that. Which there won't be when following this guide to connect from the extension to the container we run with docker in the earlier step.

```suggestion
vectorize.embedding_service_url = 'http://localhost:3000/v1/embeddings'
```